### PR TITLE
Expose wasm readiness analysis in the report (without any score)

### PR DESCRIPTION
**Pull request by isoos**
- https://github.com/dart-lang/pub-dev/issues/6785
- also did a refactor that may be one step towards some consolidation between tagging and reporting

### CODE REVIEW

File: lib/src/report/multi_platform.dart [hunk 1]
```diff
@@ -117,11 +117,47 @@ Future<ReportSection> multiPlatform(String packageDir, Pubspec pubspec) async {
...
-      basePath: packageDir,
-      subsections: [subsection],
+      basePath: context.packageDir,
+      subsections: [subsection, wasmSubsection],
       maxIssues: 20);
 }
+
+Future<Subsection> _createWasmSubsection(PackageContext context) async {
+  final tr = await context.staticAnalysis;
+  final description = 'WASM readiness';
```
Comment (sigurdm) [inline lib/src/report/multi_platform.dart:133]: Perhaps:
```suggestion
  final description = 'WASM compatibility';
```

File: lib/src/report/multi_platform.dart [hunk 2]
```diff
@@ -117,11 +117,47 @@ Future<ReportSection> multiPlatform(String packageDir, Pubspec pubspec) async {
...
+      0,
+      0,
+      ReportStatus.failed,
+    );
+  }
+
+  if (tr.tags.contains(PanaTags.isWasmReady)) {
+    return Subsection(
+      description,
+      [RawParagraph('Package is compatible with runtime `wasm`.')],
```
Comment (sigurdm) [inline lib/src/report/multi_platform.dart:149]: Do we have somewhere to link to?
Perhaps:
```suggestion
      [RawParagraph('Package is compatible with runtime `wasm`. See <https://dart.dev/web/wasm> for details.')],
```

File: lib/src/report/multi_platform.dart [hunk 3]
```diff
@@ -117,11 +117,47 @@ Future<ReportSection> multiPlatform(String packageDir, Pubspec pubspec) async {
...
+      0,
+      0,
+      ReportStatus.passed,
+    );
+  } else {
+    return Subsection(
+      description,
+      [RawParagraph('Unable to detect compatibility with runtime `wasm`.')],
```
Comment (sigurdm) [inline lib/src/report/multi_platform.dart:157]: And the same link here.

File: test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.1.json_report.md [hunk 4]
```diff
@@ -78,6 +78,10 @@ Could not determine supported platforms as package resolution failed.
 Run `dart pub get` for more information.
 </details>
 
+### [x] 0/0 points: WASM readiness
+
+Unable to detect compatibility with runtime `wasm`.
```
Comment (sigurdm) [inline test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.1.json_report.md:83]: We need a reason here. Is it because the resolution failed?

Comment (isoos) [reply test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.1.json_report.md:83]: Either the resolution failed or there are error-level issues in the code, which prevent static analysis. As these are exposed at the other parts of the report, I don't think we need to repeat them here.

**Comment by kevmoo** [comment]
Love this! thanks!

**Review by sigurdm — APPROVED** [review]

**Review by jonasfj — APPROVED** [review]

**Comment by isoos** [comment]
Could somebody push a new release with this? @sigurdm @jonasfj

**Pull request merged**